packages/framework/framework/src/compressible.test.tsx

```diff
@@ -0,0 +1,129 @@
+import { TextProps, TextStyle, Text, View } from 'react-native';
+import { compressible } from './compressible';
+import { Theme } from '@fluentui-react-native/theme-types';
+import { mergeStyles } from '@fluentui-react-native/merge-props';
+import { useSlot, withSlots } from '@fluentui-react-native/use-slot';
+import * as React from 'react';
+import toJson from 'enzyme-to-json';
+import { mount } from 'enzyme';
+
+type Variant = 'normal' | 'header' | 'caption';
+interface VariantTextTokens {
+  color?: string;
+  fontSize?: TextStyle['fontSize'];
+  fontWeight?: TextStyle['fontWeight'];
+  defaultType?: Variant;
+  header?: VariantTextTokens;
+  caption?: VariantTextTokens;
+}
+
+type VariantTextProps = TextProps & { variant?: Variant };
+
+const useVariantTokens = buildUseTokens<VariantTextTokens>({
+  color: 'black',
+  fontSize: 12,
+  fontWeight: '500',
+  defaultType: 'normal',
+  header: {
+    color: 'blue',
+    fontSize: 16,
+    fontWeight: '700',
+  },
+  caption: {
+    color: 'gray',
+    fontWeight: '300',
+  },
+});
+
+const VariantText = compressible<VariantTextProps, VariantTextTokens>(
+  (props: VariantTextProps, useTokens: UseTokens<VariantTextTokens>) => {
+    // fake theme here
+    const theme: Theme = {} as Theme;
+    // get the tokens
+    let [tokens, cache] = useTokens(theme);
+    // split the props, defaulting the variant type to the default type from the tokens
+    const { variant = tokens.defaultType, style, ...rest } = props;
+    // now apply the alternate layer tokens as appropriate
+    [tokens, cache] = applyTokenLayers(tokens, ['normal', 'header', 'caption'], cache, (layer) => layer === variant);
+    // merge styles together with what is passed in
+    const mergedStyle = mergeStyles({ color: tokens.color, fontSize: tokens.fontSize, fontWeight: tokens.fontWeight }, style);
+    // now get the slot
+    const InnerText = useSlot<TextProps>(Text, { ...rest, style: mergedStyle });
+
+    return (extra: VariantTextProps, children: React.ReactNode) => {
+      return <InnerText {...extra}>{children}</InnerText>;
+    };
+  },
+  useVariantTokens,
+);
+
+const SuperHeader = VariantText.customize({ header: { fontSize: 24, color: 'purple' } });
+
+type LabelProps = {
+  headerText: string;
+  captionText?: string;
+  headerSlot?: React.FunctionComponent<VariantTextProps>;
+  captionSlot?: React.FunctionComponent<VariantTextProps>;
+};
+
+type LabelTokens = {
+  headerVariant?: Variant;
+  captionVariant?: Variant;
+};
+
+const useLabelTokens = buildUseTokens<LabelTokens>({
+  headerVariant: 'header',
+  captionVariant: 'caption',
+});
+
+const Label = compressible<LabelProps, LabelTokens>((props: LabelProps, useTokens: UseTokens<LabelTokens>) => {
+  const theme: Theme = {} as Theme;
+  const [tokens] = useTokens(theme);
+  const { headerText, captionText, headerSlot, captionSlot } = props;
+  const Header = useSlot<VariantTextProps>(headerSlot || VariantText, { variant: tokens.headerVariant });
```

Comment:
Note that while this is allowing plugging the slot via props it could easily be done via tokens as well. It can also be completely internal. It's up to the author to determine this.